src/PowerShellEditorServices/Analysis/AnalysisService.cs

```diff
@@ -333,37 +333,6 @@ public void Dispose()
             }
         }
 
-        private static PSModuleInfo FindPSScriptAnalyzerModule(ILogger logger)
-        {
-            using (var ps = System.Management.Automation.PowerShell.Create())
-            {
-                ps.AddCommand("Get-Module")
-                  .AddParameter("ListAvailable")
-                  .AddParameter("Name", "PSScriptAnalyzer");
-
-                ps.AddCommand("Sort-Object")
-                  .AddParameter("Descending")
-                  .AddParameter("Property", "Version");
-
-                var psModuleInfo = ps.Invoke<PSModuleInfo>().FirstOrDefault();
-                if (psModuleInfo != null)
-                {
-                    logger.Write(
-                        LogLevel.Normal,
-                            string.Format(
-                                "PSScriptAnalyzer found at {0}",
-                                psModuleInfo.Path));
-
-                    return psModuleInfo;
-                }
-
-                logger.Write(
-                    LogLevel.Normal,
-                    "PSScriptAnalyzer module was not found.");
-                return null;
-            }
-        }
-
```

Comment:
Maybe I'm missing something but the deleted function seems to be replicating the Import-Module logic for loading a module.  That is, ipmo always loads the latest version in the PSModulePath anyway.  Unless the caller specifically needs the PSModuleInfo for some reason other than Import-Module.

Reply:
@rkeithhill Actually, ipmo does not load the latest version, it loads the first version it finds. This could be a bug in PowerShell Core, but here's what I did:

```powershell
PS > $env:PSModulePath
/Users/tylerleonhardt/.local/share/powershell/Modules:/usr/local/share/powershell/Modules:/usr/local/microsoft/powershell/6.0.2/Modules

## ORDER:
# 1:    /Users/tylerleonhardt/.local/share/powershell/Modules
# 2:   /usr/local/share/powershell/Modules
# 3:   /usr/local/microsoft/powershell/6.0.2/Modules

# What's in #1's Pester
PS > gci /Users/tylerleonhardt/.local/share/powershell/Modules/Pester


    Directory: /Users/tylerleonhardt/.local/share/powershell/Modules/Pester


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----           2/9/18   4:14 PM                4.1.1

# What's in #2's Pester
PS > gci /usr/local/share/powershell/Modules/Pester/


    Directory: /usr/local/share/powershell/Modules/Pester


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----          4/13/18  12:20 PM                4.3.1


# Load Pester - see it's 4.1.1
PS > ipmo Pester
WARNING: OS Information retrieval is not possible, reports will contain only partial system data
/Users/tylerleonhardt/Desktop/Tech/PowerShell/vscode/PowerShellEditorServices [master ≡]> Get-Module Pester

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     4.1.1      Pester                              {Add-AssertionOperator, AfterAll, AfterEach, AfterEachFeature...}
```

Reply:
Well that is somewhat disappointing behavior.  But it would explain why we had that method.  Another option would be to import with `-MinimumVersion` but that might be a pain to keep current.  Anytime we used a new PSSA feature we'd have to update the `MinimumVersion` parameter.

Reply:
Agreed - AFAIK that's how "PATH" works too.

> Another option would be to import with -MinimumVersion but that might be a pain to keep current

I like this idea. I don't see crazy PSSA changes just for PSES coming our way in the near future and I don't think they'll happen often. It's a small change to make us that much more resilient against someone who still has PSSA 1.0 or something that doesn't work with PSES.

@rjmholt sorry we keep giving you more work!

Reply:
I just remembered we're not actually doing `Import-Module` it's the `sessionState.ImportPSModule` and there doesn't really seem to be a good way to specify `MinimumVersion` in its overloads so I've fine with doing what we have in this PR.